.github/workflows/balena-ci.yml

```diff
@@ -8,19 +8,40 @@ env:
   ENVIRONMENT: balena-cloud.com
 
 jobs:
-  balena-cloud-build:
+  deploy-rigs:
     runs-on: ubuntu-20.04
     strategy:
       fail-fast: true
       matrix:
+        # functionally the same as leviathan-worker-armv7hf block
+        # do we need to push to fleets and blocks?
         fleets: [balena/testbot-rig, balena/testbot-rig-beta, balena/testbot-personal]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           submodules: true
-      - uses: balena-io/deploy-to-balena-action@v0.10.7
+      - uses: balena-io/deploy-to-balena-action@v0.11.0
         with:
-          balena_token: ${{ secrets.BALENA_TOKEN }}
+          balena_token: ${{ secrets.BALENA_API_KEY_PUSH }}
           fleet: ${{ matrix.fleets }}
           versionbot: true
           environment: ${{ env.ENVIRONMENT }}
+
+  deploy-blocks:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: true
+      matrix:
+        arch: [amd64, aarch64, armv7hf]
+        block: [worker, client, core]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - uses: balena-io/deploy-to-balena-action@v0.11.0
+        with:
+          balena_token: ${{ secrets.BALENA_API_KEY_PUSH }}
+          fleet: balena/leviathan-${{ matrix.block }}-${{ matrix.arch }}
+          versionbot: true
+          environment: ${{ env.ENVIRONMENT }}
+          source: ./${{ matrix.block }}
```

Comment:
This has the downside of not using the top-level version, balena.yml, and package.json for identification.

A better approach would be to merge core & client, and move worker to it's own repo as a standalone product?